### PR TITLE
Fix#5481

### LINKED ISSUE #5481: [QT5CPP] Enhancement : allow to configure api and model namespace for generated code
##### Description

It is currently not possible to configure namespace in which model and API code are generated. 
The hardcoded namespace in which the code reside is 
```
namespace Swagger{
```

This is problematic when you have multiple generated code for different APIs

##### Swagger-codegen version

2.2.2

##### Command line used for generation

done through gradle swagger plugin instead of command line. Here is the section
```
...
project('swagger-qt5cpp') {
    apply plugin: 'org.hidetake.swagger.generator'
    dependencies {
        swaggerCodegen "io.swagger:swagger-codegen-cli:2.2.2"
    }
    generateSwaggerCode {
        language = 'qt5cpp'
        inputFile = file('../src/model/rest-api/swagger.yaml')
        configFile = file('../src/model/rest-api/config-qt5cpp.json')

        dependsOn clean
    }
}
...

the config-qt5cpp.json is empty

```
##### Suggest a Fix

Have similar approach than in the cppRest code generation with following 
```
CONFIG OPTIONS
	modelPackage
	    C++ namespace for models (convention: name.space.model). (Default: io.swagger.client.model)

	apiPackage
	    C++ namespace for apis (convention: name.space.api). (Default: io.swagger.client.api)
```

### PR DESCRIPTION
**Pull request by sabras75**
### Description of the PR
fix #5481 

Add _ccpNamespace_ option to allow configurationof the namespace in which generated classes/methods reside in. 
Default is Swagger to stay compatible with previously hardcoded value.

CONFIG OPTIONS
    cppNamespace
	    C++ namespace (convention: name::space::for::api). (Default: Swagger)

### CODE REVIEW

**Comment by sabras75** [comment]
oups... just wait for now before merging, I have one last check to make.

sorry.

**Comment by wing328** [comment]
@sabras75 please take your time.

**Comment by sabras75** [comment]
Now it's good to go.

File: modules/swagger-codegen/src/main/resources/qt5cpp/api-header.mustache [hunk 1]
```diff
@@ -9,7 +9,9 @@
 
 #include <QObject>
 
-namespace Swagger {
+{{#cppNamespaceDeclarations}}
+namespace {{this}} {
+{{/cppNamespaceDeclarations}}
```
Comment (wing328) [inline modules/swagger-codegen/src/main/resources/qt5cpp/api-header.mustache]: @sabras75 Given that the original name spec is Swagger, do we still need the {{#cappNamespaceDeclarations}} .. {{/cppNamespaceDeclarations}} block? I want to keep `namespace Swagger` in the output code if no c++ namespace is provided to make the change backward compatible.

Comment (wing328) [inline modules/swagger-codegen/src/main/resources/qt5cpp/api-header.mustache]: Actually let me run a test to confirm this...

Comment (wing328) [inline modules/swagger-codegen/src/main/resources/qt5cpp/api-header.mustache]: I did a test and it's default to "Swagger", which is good 👍

**Comment by wing328** [comment]
PR merged into master. Thanks for your contribution.

**Pull request merged**